BaseClasses.py

```diff
@@ -767,6 +767,18 @@ def item_count(self, item: str, player: int) -> int:
         Utils.deprecate("Use count instead.")
         return self.count(item, player)
 
+    def has_list(self, items: Iterable[str], player: int, count: int = 1) -> bool:
```

Comment:
This behavior doesn't really match what I would expect from this function name.
With this name, I would expect it to make sure I have everything in the list.

maybe `has_from_list` (and `count_from_list`)?

Reply:
What do you think about the existing `has_group` and `count_group` that do the same thing as these new functions?

Reply:
Oh ftr I agree with you, but I'm not sure if this is a can I wanna open

Reply:
Yeah, I didn't think about comparing it with `has_group` and `count_group` from the same perspective.
It does seem like the same reasoning would apply to those, but it's also more troublesome to change them.

At least if they don't change, I think it would be good for them (all 4) to have doc strings that make it clear that it's not the other interpretation.

Reply:
I added some docstrings

Reply:
I think `has_from_list` is either easier to understand, or if not, it would at least give more reason to actually look at the doc string.

Typing-wise I'd have to read up on, but maybe Container is more accurate than Iterable, because we only need the `in` operator (`__contains__` method).

Reply:
This is `in` as in `for a in b`, not `in` as in `if a in b`. Those are different operators. The former requires `Iterable`.

Reply:
Renamed to `has_from_list` and `count_from_list`

Reply:
> This is in as in for a in b, not in as in if a in b. Those are different operators. The former requires Iterable.

Yeah, I misread the code and realized this made no sense after turning off the PC for the night :sweat: